packages/replayio/src/utils/recordings/getRecordings.ts

```diff
@@ -0,0 +1,170 @@
+import assert from "assert";
+import { existsSync, readFileSync } from "fs-extra";
+import { basename } from "path";
+import { logPath } from "./config";
+import { debug } from "./debug";
+import { LocalRecording, LogEntry, RECORDING_LOG_KIND } from "./types";
+
+export function getRecordings(): LocalRecording[] {
+  const recordings: LocalRecording[] = [];
+  const idToRecording: Record<string, LocalRecording> = {};
+
+  if (existsSync(logPath)) {
+    debug("Reading recording log %s", logPath);
+
+    const log = readFileSync(logPath, "utf8");
+    const lines = log.split("\n");
+
+    debug("Found %s recording", lines.length);
+
+    const idToStartTimestamp: Record<string, number> = {};
+
+    lines.forEach(text => {
+      if (text) {
+        const entry = JSON.parse(text) as LogEntry;
+
+        debug(JSON.stringify(entry, null, 2));
+
+        switch (entry.kind) {
+          case RECORDING_LOG_KIND.addMetadata: {
+            const { id, metadata = {} } = entry;
+            const recording = idToRecording[id];
+            assert(recording, `Recording with ID "${id}" not found`);
+            if (entry.metadata?.uri) {
+              let host = metadata.uri;
+              if (host && typeof host === "string") {
+                try {
+                  const url = new URL(host);
+                  host = url.host;
+                } finally {
+                  recording.metadata.host = host;
+                }
+              }
```

Comment:
I think I understand how you have used `finally` here but it looked a little bit tricky and took me a moment to recheck this control flow. Perhaps it would be more clear with a more standard try/catch?
```suggestion
            if (metadata?.uri) {
              if (typeof metadata?.uri === "string") {
                try {
                  const url = new URL(metadata?.uri);
                  recording.metadata.host = url.host;
                } catch (e) {
                  recording.metadata.host = metadata.uri;
                }
              }
```

Reply:
Yeah! Haha. I thought it looked weird when I went back and read it as well, so I changed it in the next PR already.

```
let host = metadata.uri;
if (host && typeof host === "string") {
  try {
    recording.metadata.host = new URL(host).host;
  } catch (error) {
    recording.metadata.host = host;
  }
}
```